src/Middleware/AuthMiddleware.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Implements the AuthMiddleware middleware class
+ *
+ * PHP Version 7
+ *
+ * @category Router
+ * @package  Middleware
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\Middleware;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Psr\Http\Message\URIInterface;
+use \Psr\Http\Server\MiddlewareInterface;
+use \Psr\Http\Server\RequestHandlerInterface;
+use \LORIS\Http\StringStream;
+
+/**
+ * AuthMiddleware is a type of Middleware which determines whether or
+ * not a request is properly authenticated, and returns a 403 if not.
+ *
+ * @category Router
+ * @package  Middleware
+ * @author   Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AuthMiddleware implements MiddlewareInterface, MiddlewareChainer
```

Comment:
I can't find where MiddlewareChainerMixin is defined

Reply:
I just added it to the PR